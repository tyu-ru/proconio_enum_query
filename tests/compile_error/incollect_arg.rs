#[proconio_enum_query::proconio_enum_query(hoge)]
enum Query1 {
    A,
}

#[proconio_enum_query::proconio_enum_query(fuga(piyo))]
enum Query2 {
    A,
}

#[proconio_enum_query::proconio_enum_query(foo = 10)]
enum Query3 {
    A,
}

#[proconio_enum_query::proconio_enum_query(start_index = abc)]
enum Query3 {
    A,
}

#[proconio_enum_query::proconio_enum_query(start_index = "abc")]
enum Query3 {
    A,
}

fn main() {}
