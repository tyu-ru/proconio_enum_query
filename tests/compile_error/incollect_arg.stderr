error: expected `=`
 --> $DIR/incollect_arg.rs:1:1
  |
1 | #[proconio_enum_query::proconio_enum_query(hoge)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected `=`
 --> $DIR/incollect_arg.rs:6:48
  |
6 | #[proconio_enum_query::proconio_enum_query(fuga(piyo))]
  |                                                ^^^^^^

error: unknown args
  --> $DIR/incollect_arg.rs:11:44
   |
11 | #[proconio_enum_query::proconio_enum_query(foo = 10)]
   |                                            ^^^

error: expected literal
  --> $DIR/incollect_arg.rs:16:58
   |
16 | #[proconio_enum_query::proconio_enum_query(start_index = abc)]
   |                                                          ^^^

error: must be isize literal
  --> $DIR/incollect_arg.rs:21:58
   |
21 | #[proconio_enum_query::proconio_enum_query(start_index = "abc")]
   |                                                          ^^^^^
